broker/config/settings.yml

```diff
@@ -430,6 +431,12 @@ defaults: &defaults
     skip_ssl_validation: true
     default_task_poll_interval: 10000
     lock_deployment_max_duration:  86400 # lock ttl for deployments in seconds. Its kept at 24 hrs - 60 * 60 * 24
+    max_workers: 24
+    policies:
+      user:
+        max_workers: 50
+      scheduled:
+        max_workers: 50
```

Comment:
Currently the approach is to give 50 percent workers each to user triggered and scheduled task. But this approach still may have a limitation on question of fare share for various request served by SF. For example if a user triggers 12 long running upgrades, followed by a create. This will result in create being blocked until all the upgrades are completed.

Reply:
Yes but that's the problem of BOSH having limited resources. True fair share cannot be achieved either way

Reply:
@ashishjain14 @kamath-prasad @abh1kg 
I thought the original intention of the problem was to address what ashish mentioned in the above. Else what is the use of queuing in SF Broker side, we might as well let it queue on BOSH. Dont see the real benefit then?

My understanding was, if there are 24 workers then say 50-70% of it would be max used by major version upgrade. So at any point in time, then we must check for processing/queued tasks and if their combined value is less than the configured percentage only then we push that request to BOSH. Else queue it. As an example, 
out of 24 workers, assume we say 20 workers can be used for major version update and rest 4 workers we leave it free for any other adhoc operation that comes in.

Now before pushing a major version upgrade we see how many tasks are in processing/queued. If this count is less than 20, then we let this upgrade to bosh, else queue it. This way, we will push major version upgrade only when we feel it does not hog BOSH completely and it leaves for other operations also to take place.

This is the main use case which mongodb wants us to solve. May be please double check with MongoDB team.